### ISSUE TITLE
May want regular file integrity checks during a backfill for the s3 client.

### ISSUE THREAD

**Issue by mpawliszyn** [cashapp/backfila#312]
There are several assumptions around the files not changing out from under the s3 client. We might want to check this for a better customer experience.
https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html

We probably need some kind of side channel data for this. The prepare can pull this information perhaps?

**Comment by shellderp** [comment]
You could get s3 modified timestamp, return it as a parameter from Prepare, then compare the current value on each operation, failing if it has changed. Prepare is allowed to create new parameters dynamically.

**Comment by mpawliszyn** [comment]
Yeah that is a good idea both on the timestamp and on the parameter
